libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -922,6 +1024,69 @@ string YulUtilFunctions::updateStorageValueFunction(Type const& _type, boost::op
 	});
 }
 
+string YulUtilFunctions::writeToMemoryFunction(Type const& _type)
+{
+	if (auto ref = dynamic_cast<ReferenceType const*>(&_type))
+	{
+		solAssert(
+			ref->location() == DataLocation::Memory,
+			"Can only update types with location memory."
+		);
+
+		return writeToMemoryFunction(*TypeProvider::uint256());
+	}
+
+	string const functionName =
+		string("update_memory_value_") +
+		_type.identifier();
+
+	return m_functionCollector->createFunction(functionName, [&] {
+		if (/*auto str = */dynamic_cast<StringLiteralType const*>(&_type))
+		{
+			/*m_context << Instruction::DUP1;
+			storeStringData(bytesConstRef(str->value()));
+			if (_padToWordBoundaries)
+				m_context << u256(max<size_t>(32, ((str->value().size() + 31) / 32) * 32));
+			else
+				m_context << u256(str->value().size());
+			m_context << Instruction::ADD;*/
+			solUnimplemented("String literals not yet implemented!");
+		}
+		else if (
+			_type.category() == Type::Category::Function &&
+			dynamic_cast<FunctionType const&>(_type).kind() == FunctionType::Kind::External
+		)
+		{
+			return Whiskers(R"(
+				function <functionName>(memPtr, selector, addr) {
```

Comment:
Is the order of `selector` and `addr` correct here?

Reply:
well, it's not used anywhere yet, so the order can be whatever we want it to be. Does addr first make more sense?

Reply:
It should be the same order as the one used by the other functions that deal with function types.

Reply:
Alright, I adjusted it in my last push already to match that